Add man page generation, change man section, update docs to reflect name change, and couple other tweaks

jlinton opened the pull request:
Some distro's prefer man pages over .html or .rst files dumped into documentation directories. SPHINX can generate a man page from the existing documentation. So let's add a cmake rule to do it, and adjust the man section to 7.

Also, when reading the resulting documentation, there are a number of problems with it. Starting with the fact its out of date WRT to the hardware/etc requirements, build instructions, etc. Its also still calls itself "Hyperscan"

Let's do a quick pass to fix all the glaring errors.

jlintonarm committed:
documentation: Add cmake option to build man pages

Man pages tend to be preferred in some circles, lets add an
option to build the vectorscan documentation that way.

Signed-off-by: Jeremy Linton <jeremy.linton@arm.com>

jlintonarm committed:
documentation: Update project name and copyright

The project name in the documentation should probably
be updated to reflect that this is vectorscan. Update
the copyright too.

Signed-off-by: Jeremy Linton <jeremy.linton@arm.com>

jlintonarm committed:
documentation: Replace project name with Vectorscan and general updates

The generated documentation continues to refer to Hyperscan
despite the project now being VectorScan. Lets replace many
of the Hyperscan references with Vectorscan.

At the same time, lets resync the documentation here with the
vectorscan readme. This updates the supported platforms/compilers
and build options.

Signed-off-by: Jeremy Linton <jeremy.linton@arm.com>

markos commented on doc/dev-reference/conf.py.in:
```diff
@@ -43,7 +43,7 @@ source_suffix = '.rst'
 
 # General information about the project.
-project = u'Hyperscan'
+project = u'Vectorscan'
```
such changes need checking, for a start, we cannot change the project name like that, and we need to add the relevant copyright for our company VectorCamp for the years 2020-2024. In general references to the project can be changed, but when it comes to copyright mentions, I will need to look over them. In fact, I would put everywhere possible that this project is a fork of Hyperscan, the original Intel project.

jlinton replied:
INAL, but I don't understand the part about not changing the project name. It's already been changed, no? So it probably needs to stop calling itself 'hyperscan' (is that an intel trademark?)

I didn't want to start going down this rabbit hole, but the man pages were coming out as hyperscan.7 and saying hyperscan everywhere rather than vectorscan. 

But I'm fine with all the rest of the review comments, it's just this one I don't fully understand. 


PS: I also should have updated the pkgconfig description, although for API/ABI compatibility, i'm assuming that has to remain as libs.pc.

jlinton replied:
Also, I probably won't update this till tue, since its a long weekend.

markos replied:
I should be clearer, I meant the copyright below. For example, Hyperscan is Intel copyright (years up to current), Vectorscan is Intel copyright (years up to 2020, that's when we forked) AND VectorCamp PC 2020-2024 and in some files the respective copyrights of some contributors for some patches, eg. Arm Ltd. These statements have to be fixed.

markos replied:
But also IANAL, need to consult some people on that to be totally correct.

markos commented on doc/dev-reference/getting_started.rst:
```diff
@@ -41,9 +41,9 @@ Very Quick Start
      * ``MsBuild.exe`` --- use Visual Studio MsBuild
```
Since MS Windows support has been completely removed since a few releases ago, I would also remove mentions to Windows builds.

jlinton replied:
Should be gone.

markos commented on doc/dev-reference/getting_started.rst:
```diff
@@ -32,7 +32,7 @@ Very Quick Start
    Generators that might work include:
       * ``Xcode`` --- OS X Xcode projects.
```
Xcode support has not been verified to work, and is not a supported, I would probably remove it or add a mention that it's an unsupported configuration.

jlinton replied:
Noted it is unsupported.

markos commented on doc/dev-reference/getting_started.rst:
```diff
@@ -55,20 +55,22 @@ Requirements
     * Intel Advanced Vector Extensions 2 (Intel AVX2)
+    * Arm NEON
+    * Arm SVE and SVE2
+    * Arm SVE2 BITPERM
```
+ IBM Power8/Power9 VSX

jlinton replied:
Sure

markos commented on doc/dev-reference/getting_started.rst:
```diff
@@ -79,22 +81,23 @@ These can be determined at library compile time, see :ref:`target_arch`.
 
     * GCC, v4.8.1 or higher
     * Clang, v3.4 or higher (with libstdc++ or libc++)
     * Intel C++ Compiler v15 or higher
```
Intel and Visual Studio are not supported compilers, they need to be removed.

markos commented on doc/dev-reference/getting_started.rst:
```diff
@@ -79,22 +81,23 @@ These can be determined at library compile time, see :ref:`target_arch`.
+modern C and C++ compiler -- in particular, Vectorscan requires C99 and C++11
 compiler support. The supported compilers are:
 
     * GCC, v4.8.1 or higher
```
GCC v9 and Clang v5 are now the default minimum as C++17 is required.

jlinton replied:
Sure

markos commented on doc/dev-reference/getting_started.rst:
```diff
@@ -79,22 +81,23 @@ These can be determined at library compile time, see :ref:`target_arch`.
 
 Linux:
 
 * Ubuntu 14.04 LTS or newer
```
+Debian 10
+Ubuntu 20.04 LTS

jlinton replied:
Should be done.

markos commented on doc/dev-reference/getting_started.rst:
```diff
@@ -168,10 +171,10 @@ Common options for CMake include:
 |                        | the default static library.                        |
 +------------------------+----------------------------------------------------+
-| BUILD_STATIC_AND_SHARED| Build both static and shared Hyperscan libs.       |
+| BUILD_STATIC_AND_SHARED| Build both static and shared Vectorscan libs.      |
```
could you please amend this according to the options in the README/wiki ?

jlinton replied:
This should be done

markos commented on doc/dev-reference/getting_started.rst:
```diff
@@ -254,11 +257,11 @@ resolved so that the right version of each API function is used. There is no
 by the ELF loader once when the binary is loaded.
 
-If the Hyperscan library is used on x86 systems without ``SSSE3``, the runtime
+If the Vectorscan library is used on x86 systems without ``SSSE3``, the runtime
```
changed to SSE4.2

markos commented on doc/dev-reference/tools.rst:
```diff
@@ -56,7 +56,7 @@ produce output like this::
 
     Signatures:        /tmp/patterns
-    Hyperscan info:    Version: 4.3.1 Features:  AVX2 Mode: STREAM
+    Vectorscan info:    Version: 4.3.1 Features:  AVX2 Mode: STREAM
```
Version should be changed to latest.

jlinton replied:
I ended up fixing hsbench to actually report Vectorscan as well.

jlinton commented on doc/dev-reference/CMakeLists.txt:
```diff
@@ -32,4 +33,14 @@ add_custom_target(dev-reference
+        "${CMAKE_CURRENT_SOURCE_DIR}"
+        "${SPHINX_MAN_DIR}"
+    DEPENDS dev-reference-doxygen
+    COMMENT "Building HTML dev reference with Sphinx")
```
Probably should say "man" here.

jlintonarm committed:
pkgconfig: Correct library description

Correct the description in the pkgconfig file, but
leave the name alone as we want to remain compatible
with projects utilizing hyperscan.

Signed-off-by: Jeremy Linton <jeremy.linton@arm.com>

jlintonarm committed:
hsbench: Update test program output

While fixing the documentation, it was noticed that the hsbench
output was still referring to the project as Hyperscan.
Lets correct it.

Signed-off-by: Jeremy Linton <jeremy.linton@arm.com>

jlinton commented:
All these have been fixed IMHO. None of this is 100%, as there remains a number of hyperscan references that should be updated, but it is a bit of a rabbit hole for fixing the one thing I want to correct, which is the generation of the man pages.

I've also updated the pkgconfig to reflect it is no longer hyperscan despite keeping the hslib variable to maintain build compatibility with hyperscan users.

markos commented on doc/dev-reference/conf.py.in:
```diff
@@ -43,8 +43,8 @@ source_suffix = '.rst'
-project = u'Hyperscan'
-copyright = u'2015-2018, Intel Corporation'
+project = u'Vectorscan'
+copyright = u'2015-2018, Intel Corporation; 2020-2024, VectorCamp; and other contributors'
```
should be -2020 here for Intel.

jlinton replied:
It should be updated.

jlinton commented:
Somehow, I appeared to have picked up the develop branch with differing commit IDs, which should now be fixed. However, I wonder if it was some kind of GitHub display glitch now. Either way, it looks right.

markos reviewed (approved):
LGTM, Approved
